client.js

```diff
@@ -362,25 +361,43 @@ DHT.prototype._put = function (opts, cb) {
   function put (node) {
     if (node.data) return // skip data nodes
     pending += 1
-    var t = self._getTransactionId(node.addr, next(node))
-    var data = {
+
+    var t = self._getTransactionId(node.addr, putOnGet)
+    self._send(node.addr, {
       a: {
-        id: opts.id || self.nodeId,
-        v: opts.v
+        id: self.nodeId,
+        target: hash
       },
       t: transactionIdToBuffer(t),
       y: 'q',
-      q: 'put'
-    }
-    if (isMutable) {
-      data.a.token = opts.token || self._generateToken(node.addr)
-      data.a.seq = opts.seq
-      data.a.sig = opts.sign(encodeSigData(opts))
-      data.a.k = opts.k
-      if (opts.salt) data.a.salt = opts.salt
-      if (opts.cas) data.a.cas = opts.cas
+      q: 'get'
+    })
+
+    function putOnGet (err, res) {
```

Comment:
in the bep44 spec they actually don't document this in the request/response payloads, but you have to do a get first to each node, and that node returns you a write token. the write tokens are required for both mutable and immutable